python/sglang/srt/function_call/pythonic_detector.py

```diff
@@ -32,13 +32,24 @@ def __init__(self):
             re.DOTALL,
         )
 
+    @staticmethod
+    def _text_strip(text: str) -> str:
+        # Llama 4 model sometime will output <|python_start|> and <|python_end|> tokens
+        # remove those tokens
+        text = text.replace("<|python_start|>", "")
+        text = text.replace("<|python_end|>", "")
+        return text
+
     def has_tool_call(self, text: str) -> bool:
-        return bool(self.tool_call_regex.search(text.strip()))
+        return bool(self.tool_call_regex.match(self._text_strip(text.strip())))
```

Comment:
We should not use match here. This will make it failed to check if there is a mixed text with the function call.

`match` vs `search`
`match()`: Only matches if the regex pattern matches from the very beginning of the string
`search()`: Finds the pattern anywhere in the string